content/en/docs/Concepts/DTO.md

```diff
@@ -1,11 +1,122 @@
+
+### - What is overposting?
+
+Overposting is a type of cyber attack that targets websites. More specifically, overposting targets models that expect an input from a user, but have other variables that are intentionally hidden from the user. For example, if a model has a “IsAdmin” variable that limits what a user can and cannot do, a hacker who knows this information can manually add an extra field to an input that would allow them to change the “IsAdmin” variable.
+
+### - How is this prevented?
+
+Take a look at an example set of code \(provided by [Scott Hanselmen](https://www.hanselman.com/blog/aspnet-overpostingmass-assignment-model-binding-security)\):
+
+```
+public class Person
+{
+    public int ID { get; set; }
+    public string First { get; set; }
+    public string Last { get; set; }
+    public bool IsAdmin { get; set; }
+}
+```
+
+```
+[HttpPost]
+[ValidateAntiForgeryToken]
+public async Task<IActionResult> Create(Person person)
+{
+```
+
+```
+ if (ModelState.IsValid)
+    {
+        _context.Add(person);
+        await _context.SaveChangesAsync();
+        return RedirectToAction("Index");
+    }
+    return View(person);
+}
+```
+
+Here we see an entity and a corresponding action that accepts the entity as a parameter. Let's say that the controller only wants the “First” and “Last” variables to be gathered from an input. A hacker can very easily add another field to an input that allows them to edit either the “ID” or the “IsAdmin” field. Now, look at the same code again but this time it is using binding.
+
+```
+[HttpPost]
+[ValidateAntiForgeryToken]
+public async Task<IActionResult> Create([Bind("First,Last")] Person person)
+{
+    if (ModelState.IsValid)
+    {
+        _context.Add(person);
+        await _context.SaveChangesAsync();
+        return RedirectToAction("Index");
+    }
+    return View(person);
+}
+```
+
+Now the action has bound the desired variables to itself. This means that it will only accept inputs that edit either the “First” or “Last” variables and will not take any inputs from anything else. This makes the other variables such as “IsAdmin” completely inaccessible to hackers.
+
+## Notes from Microsoft
+
+Microsoft recognizes DTO’s as being superior to binding in terms of security. Take note of these resources for more information on DTO’s and their security in terms of overposting: [Create Data Transfer Objects](https://docs.microsoft.com/en-us/aspnet/web-api/overview/data/using-web-api-with-entity-framework/part-5), [Microsoft Security note about Overposting](https://docs.microsoft.com/en-us/aspnet/core/data/ef-mvc/crud?view=aspnetcore-6.0#security-note-about-overposting)
+
+## Examples
+
+Below is an example on a Class and a corresponding DTO that limits what parts of the class are visible \([Baeldung](https://www.baeldung.com/java-dto-pattern)\). For an example of regular binding see the “**How is this prevented?**” section of this document.
```

Comment:
No need for the in-text citation, as long as you reference it properly at the end. I checked and that sentence isn't in the article, so you should be good without the in-text citation.

Reply:
Insert a comma after "For an example"